ci_tools/git_evaluation_tools.py

```diff
@@ -423,6 +424,16 @@ def check_previous_contributions(repo, author):
 
     with subprocess.Popen(cmds, stdout=subprocess.PIPE) as p:
         result, err = p.communicate()
+        returncode = p.returncode
     print(err)
+    print(returncode)
+    if returncode:
+        while returncode:
+            time.sleep(10)
```

Comment:
Hi @EmilyBourne. Why is it necessary to wait 10 seconds here?

Reply:
It only fails when there are too many requests. The github message asks you to wait "a few seconds" then try again

Reply:
This is documented here
https://docs.github.com/en/rest/guides/best-practices-for-integrators?apiVersion=2022-11-28#dealing-with-secondary-rate-limits
Ideally it should be improved to collect the right wait time

Reply:
Isn't there the danger of getting stuck here if the system is down? Maybe we can return an error after a certain number of tries.

Reply:
I agree. I have capped it at 10 tries ([3781a6d](https://github.com/pyccel/pyccel/pull/1395/commits/3781a6d62b9cd19a9214e09a057bfa2a96b788cf))

I have found an example of this failure:
https://github.com/EmilyBourne/pyccel/actions/runs/5001434467/jobs/8960109824

I don't seem to have access to the `Retry-after` value or the rate limit. I imagine 10 seconds should be sufficient